### ISSUE TITLE
Mismatched type and gnsdsid

### ISSUE THREAD

**Issue by cyrusjk** [BeakeS/TeenyUbloxConnect#1]
Testing a bGeigie Zen, using an M10 module, when I call ubloxNAVSATSVInfo_t gnssIdType() and gnssId() I frequently see a type of G but an Id of 2. Based on the comments it seems like type and Id are paired, and that G should only have an Id of 0. 

Here is an example:
NavSat type:G,svId:10,gnssId:0
NavSat type:G,svId:27,gnssId:0
NavSat type:G,svId:36,gnssId:2
NavSat type:G,svId:23,gnssId:0
NavSat type:G,svId:11,gnssId:2

Is this a misinterpretation of the documentation, or an actual issue?

**Comment by robouden** [comment]
cyrus,

Thanks for finding the issue.

Rob

**Comment by cyrusjk** [comment]
I don't know if it's THE issue, but it is a point of confusion.

**Comment by robouden** [comment]
Cyrus,

You filed it as an issue..:)

**Comment by BeakeS** [comment]
Fixed this issue yesterday.  This was a gnssIdType labeling issue and does not affect GPS functionality.

For TeenyGPSTestbed_C2 - Load latest version.
For Zen - Change TeenyUbloxConnect library pointer to version 1.0.22

**Comment by cyrusjk** [comment]
Thanks for looking into that issue, and I'm glad it was more cosmetic than
practical.

That being said, the Zen I have with the M10 is not very accurate, and I am
trying to zero in on either the issue or some kind of resolution.
Currently, I have it sitting stationary, in a windowsill, so it IS
partially obscured from directly above and to the west, but the fix is
still being returned as valid, and the locations are being logged even
though they are inaccurate. PDOP is used to color, lower (lower DOP) is
green.

[image: image.png]

Even placed in full view of the sky, still stationary, it still wanders a
significant amount.
[image: image.png]

I get better accuracy out of my bGeigie nano or just about anything else,
so I am assuming that some unexploited configurations might make it better.

I'm not a c++ guy so my work isn't very efficient, but I have been adding
some M10-specific features to see if they have an effect or can give some
insight into the validity of the fix, but I've yet to find anything. If you
have any ideas I'd love to give them a shot.

**Comment by BeakeS** [comment]
I've gone down that rabbit hole of tweaking M8 and M10 ublox module settings beyond those I made available in the testbed.  (I used u-center and u-center-2 windows app from u-blox.com.)  With each attempt, I always wound up reverting back to factory default settings and just selecting the appropriate gnss for my area using the testbed.  The option to do a factory reset in the testbed is - Main Menu -> Device Settings -> GPS Settings -> Factory Reset.

I am not able to see the images in your comment.  It just shows as [image: image.png].

I am seeing ~16m wandering with the Zen (with an M10 module) placed 2 meters outside my house and ~90m with the Zen on my windowsill.
